pandas/core/groupby/grouper.py

```diff
@@ -348,9 +349,19 @@ def _set_grouper(self, obj: FrameOrSeries, sort: bool = False):
             if getattr(self.grouper, "name", None) == key and isinstance(
                 obj, ABCSeries
             ):
-                # pandas\core\groupby\grouper.py:348: error: Item "None" of
-                # "Optional[Any]" has no attribute "take"  [union-attr]
-                ax = self._grouper.take(obj.index)  # type: ignore[union-attr]
+                # Sometimes (when self.indexer is not None) self._grouper will be
+                # sorted while obj is not. In this case there is a mismatch when we
+                # call self._grouper.take(obj.index) so we need to undo the sorting
+                # before we call _grouper.take.
+                if self.indexer is not None:
+                    assert self._indexer is not None
```

Comment:
what is the point of this assert?

Reply:
This is for the same reason as the other assert you asked me to make in your last review.  If I don't make that assert I get the following error from mypy:

error: Item "None" of "Optional[Any]" has no attribute "argsort"  [union-attr]